커뮤니티 카테고리정렬, 게시글수정, 게시글댓글, 헤더검색기능 등 PULL REQUEST 리뷰 및 확인 요청드립니다.

rsvprzdz opened the pull request:
## 커뮤니티 카테고리정렬, 게시글수정, 게시글댓글, 헤더검색기능 등 PULL REQUEST 리뷰 및 확인 요청드립니다.

### 주요 커밋사항

- 커뮤니티 댓글기능
```
> 댓글 및 답글 조회, 작성, 삭제수정기능(CRUD)
> 댓글 기능 중, 아이디 태그 기능은 답글의 답글 기능이므로 구현기능에서 제외하였습니다.
> 잔여작업항목: 작성자 본인이 댓글 or 답글을 남길 경우 '글쓴이 ' 로 표기 구현
```

- 메인페이지, EXMAP
```
> MAIN: 슬라이드, TOP10의 전시/행사 목록 클릭 시 해당하는 상세페이지로 이동 처리
> EXMAP: 유저가 마커를 클릭해 띄운 모달창의 전시/행사 클릭 시 해당하는 상세페이지로 이동 처리
```

- 헤더 검색기능
```
> 유저가 입력한 키워드가 제목에 포함되어있는 전시 검색 기능
> 해당 기능을 헤더가 포함되어있는 모든 페이지에서 구현하기 위해 헤더에 jQuery 스크립트를 import했습니다.
```
> [!CAUTION]
> 헤더에 jQeury가 포함되어 각 페이지의 jQeury와 버전 충돌 이슈가 예상됩니다.
> 버전 충돌 이슈 확인을 위해 해당 커밋 merge 이후 각 페이지 기능테스트 후 팀내 공유 부탁드립니다.

> [!IMPORTANT]
> 헤더 검색기능의 경우, 정현님의 ExhibitionEvent vo객체와 exhibition controller, service, mapper에서 작업을 진행하였습니다.
> 이로 인한 conflict가 예상되오니. conflict 발생 시 컨택부탁드립니다~
> conflict 예상 요소: header.html, header.js, exhibitionMapper.xml, exhibitionService.java, exhibitionMapper.java

<br/>

- 기타사항
```
> favicon 파일 임시로 제작하여 적용해두었습니다.
> 전시/행사 목록 아이템들 이미지 높이를 통일하여 텍스트 정렬을 일치시켰습니다(강사님의견), 배경화면은 가장 옅은 프로젝트컬러 적용
```

rsvprzdz committed:
feat. 커뮤니티 목록 카테고리별 조회기능, 커뮤니티 게시글 수정기능, 커뮤니티 게시글 댓글 기능, 메인페이지 슬라이드 조회기능
design: 강사님 의견 반영하여 메인페이지 및 커뮤니티페이지 css 일부 수정(사이즈 조절 및 색상 시인성 향상)

커뮤니티 댓글기능: 댓글 및 답글 조회, 작성, 삭제기능 완료 / 수정기능 작업 예정
메인페이지: 슬라이드 아이템 클릭 시 상세정보 페이지로 이동처리 기능 작업 예정

rsvprzdz committed:
Merge branch 'main' of https://github.com/2024-TeamX/TeamX

rsvprzdz committed:
Merge branch 'main' of https://github.com/2024-TeamX/TeamX

rsvprzdz committed:
feat: 헤더 검색기능 추가, 파비콘 추가,
fix: 글작성/수정 관련 버그 수정

main.html: 더미데이터 요소 삭제
header.html: jquery 스트립트 추가

conflict 예상 요소: header.html, header.js, exhibitionMapper.xml, exhibitionService.java, exhibitionMapper.java

jeonghyunkim89 reviewed (approved):
커뮤니티 카테고리정렬, 게시글수정, 게시글댓글, 헤더검색기능 확인했습니다

hailey-516 commented on exsite/src/main/java/com/teamx/exsite/controller/community/CommunityController.java:
```diff
@@ -94,20 +90,91 @@ public String communityList(@RequestParam(value="cpage", defaultValue="1")int cu
+	public String toEditPost(@PathVariable int postNo, Model model) {
+	    // 기존 게시글 정보를 조회
+	    Board board = boardService.selectDetail(postNo);
+	    System.out.println(board);
```
테스트로 콘솔 찍어 본 코드는 주석 처리 부탁드립니다

rsvprzdz replied:
해당 컨트롤러 이외에 담당 페이지 컨트롤러 체크 후
테스트 콘솔 부분 모두 주석처리 완료하였습니다.
피드백감사합니다~

hailey-516 commented on exsite/src/main/java/com/teamx/exsite/service/community/BoardService.java:
```diff
@@ -28,8 +30,15 @@ public int selectListCount() {
+		// 예시) 
+		// offset = (현재 페이지 - 1) * 페이지 당 게시글 수 = (2 - 1) * 15 = 15
+		// limit = 15 (한 페이지에 표시할 게시글 수) => RowBounds에 매개변수로 15, 15를 전달하면 15번째 로우부터 15개 조회해서 반환해줌(레전드;;)
+		// 이모든걸 마이바티스 rowBounds가 해준다 이말씀
```
대박 나중에 다시 봐야지

hailey-516 commented on exsite/src/main/resources/static/js/community/communityPost.js:
```diff
@@ -28,4 +235,111 @@ function insertParentReply(){
+
+      // 댓글 삭제 요청
+      $.ajax({
+         url: "/community/childrenReply/delete",
```
url 요청 시 소문자로 수정 부탁드립니다.

rsvprzdz replied:
CommunityController, ExmapController,
communityPost.js, commynityList.js, exmap.js
url 맵핑 일괄 소문자로 수정 완료하였습니다~
피드백 감사합니다. (이후 동일 피드백 관련 일괄 답변)

hailey-516 commented on exsite/src/main/java/com/teamx/exsite/controller/community/CommunityController.java:
```diff
@@ -383,15 +419,68 @@ public String deleteParentReply(@RequestBody ParentReply parentReply) {
+	 * @return 삭제 성공여부
 	 */
 	@ResponseBody
 	@PostMapping("/community/childrenReply/delete")
```
url 소문자로 수정 확인 부탁드립니다.

hailey-516 commented on exsite/src/main/java/com/teamx/exsite/controller/community/CommunityController.java:
```diff
@@ -383,15 +419,68 @@ public String deleteParentReply(@RequestBody ParentReply parentReply) {
+	 * @return 수정 성공여부
+	 */
+	@ResponseBody
+	@PostMapping("/community/parentReply/edit")
```
url 소문자로 수정 확인 부탁드립니다.

hailey-516 commented on exsite/src/main/java/com/teamx/exsite/controller/community/CommunityController.java:
```diff
@@ -383,15 +419,68 @@ public String deleteParentReply(@RequestBody ParentReply parentReply) {
+	 * @return 수정 성공여부
+	 */
+	@ResponseBody
+	@PostMapping("/community/childrenReply/edit")
```
url 소문자로 수정 확인 부탁드립니다.

hailey-516 commented on exsite/src/main/java/com/teamx/exsite/controller/exmap/ExmapController.java:
```diff
@@ -44,7 +44,7 @@ public List<ExmapCoordinate> selectAllCoordinates(@RequestParam String guname) {
 	
 	
-	@GetMapping("/exmap/exhibition-info")
+	@GetMapping("/exmap/exhibitionInfo")
```
mapping 부분 요청 동일하게 소문자로 수정 확인 부탁드립니다.

대다수의 서버는 url에서 대소문자를 구분합니다.
관례상 URL을 모두 소문자로 작성하는 것이 좋습니다 (예: /exhibitioninfo), 이는 URL 관리와 일관성을 유지하는 데 유리합니다.

hailey-516 commented on exsite/src/main/resources/static/js/community/communityPost.js:
```diff
@@ -338,8 +336,136 @@ $(document).on('click', '.community-reply-delete-btn', function () {
+
+   if( $(button).siblings('.reply-input').val().trim().length > 0 ){
+      $.ajax({ 
+         url: "/community/childrenReply/edit",
```
url 확인 부탁드립니다.

rsvprzdz committed:
fix: 자식댓글 입력, 삭제 기능 관련 버그 픽스, 댓글 수 카운팅 기능 관련 버그 픽스
docs: 테스트용 콘솔 로깅 구문 일괄 주석처리
refactor: mapping url 대문자 일괄 소문자로 변경처리(카멜케이스에서 슬래시( / ) 로 구분 처리)

버그픽스내역:
 - 마지막 답글 삭제시 윗 답글까지 일괄 삭제되는 문제(mapper 수정)
 - 댓글 입력 후 입력한 부모댓글의 자식댓글만 조회되고, 다른 부모댓글의 자식댓글들이 조회되지 않는 문제(mapper 및 js 수정)

hailey-516 reviewed (approved):
커밋 내용과 변경 사항 전부 확인했습니다. 고생하셨습니다!
